strawberryfields/backends/gaussianbackend/ops.py

```diff
@@ -222,133 +139,13 @@ def xmat(n):
-    """Defines a limited sizes dictionary.
-    Used to limit the cache size.
-    """
-    def __init__(self, *args, **kwargs):
-        self.size_limit = kwargs.pop("size_limit", None)
-        OrderedDict.__init__(self, *args, **kwargs)
-        self._check_size_limit()
-
-    def __setitem__(self, key, value):
-        OrderedDict.__setitem__(self, key, value)
-        self._check_size_limit()
-
-    def _check_size_limit(self):
-        if self.size_limit is not None:
-            while len(self) > self.size_limit:
-                self.popitem(last=False)
-
-
-MAXSIZE = 1000
-Tee = tee([], 1)[0].__class__
-
-def memoized(f, maxsize=MAXSIZE):
-    """Used to memoize a generator"""
-    cache = LimitedSizeDict(maxsize=maxsize)
-    def ret(*args):
-        if args not in cache:
-            cache[args] = f(*args)
-        if isinstance(cache[args], (GeneratorType, Tee)):
-            # the original can't be used any more,
-            # so we need to change the cache as well
-            cache[args], r = tee(cache[args])
-            return r
-        return cache[args]
-    return ret
-
-
-@memoized
-def partitions(s, singles=True, pairs=True):
-    """Returns the partitions necessary to calculate click probabilities."""
-    # pylint: disable=too-many-branches
-    #print(s)
-    if len(s) == 2:
-        if singles:
-            yield (s[0],), (s[1],)
-        if pairs:
-            yield s
-    else:
-        # pull off a single item and partition the rest
-        if singles:
-            if len(s) > 1:
-                item_partition = (s[0],)
-                rest = s[1:]
-                rest_partitions = partitions(rest, singles, pairs)
-                for p in rest_partitions:
-                    if isinstance(p[0], tuple):
-                        yield ((item_partition),) + p
-                    else:
-                        yield (item_partition, p)
-            else:
-                yield s
-        # pull off a pair of items and partition the rest
-        if pairs:
-            # idx0 = 0
-            for idx1 in range(1, len(s)):
-                item_partition = (s[0], s[idx1])
-                rest = s[1:idx1] + s[idx1+1:]
-                rest_partitions = partitions(rest, singles, pairs)
-                for p in rest_partitions:
-                    if isinstance(p[0], tuple):
-                        yield ((item_partition),) + p
-                    else:
-                        yield (item_partition, p)
-
-
-def gen_indices(l):
-    """
-    Given a list of integers [i_0,i_1,...,i_m] it returns a list of int8 integers as follows
-    [0,0,...,1,1,...,2,2....,....,m,m,m...]
-    where 0 is repated i_0 times, 1 is repated i_1 times and so on and so forth
-    """
-    x = np.zeros(sum(l), dtype=np.uint8)
-    ss = 0
-    for i, ii in enumerate(l):
-        for j in range(ii):
-            x[j+ss] = i
-        ss += l[i]
-    return x
-
-
-def fock_prob(s2, ocp, tol=1.0e-13):
+def fock_prob(gaussian_state, ocp):
```

Comment:
@nquesada tests are currently failing due to this function
i) the functions ``smeanxp()`` and ``scovmatxp()`` are being called as if they were methods of `gaussian_state`. However, these functions exist in ``gaussiancircuit.py` rather than `states.py`
ii) you will need to match the names r and mu